test/opt/test_utils.cpp

```diff
@@ -55,4 +55,48 @@ TEST(JoinNonDebugInsts, Cases) {
                  "the only remaining string"}));
 }
 
+namespace {
+typedef struct {
```

Comment:
Oh, man, this is C style.

I'd suggest just doing `struct SubstringReplacementTestCase`.